### PR TITLE
#163388133 Implement user email verification

### PR DESCRIPTION
**Pull request by PaulinhoNdegwa**
#### What does this PR do?
- Customize the API's landing page to suit an API style
- Added an is_verified field on User model
- Create a HTML template for email message
- Add code to draft and send email
- Add code to verify user token from verification link

#### Description of Task to be completed
- Customize the API's landing page to suit an API style
- Added an is_verified field on User model
- Create a HTML template for email message
- Add code to draft, send and verify email

#### How should this be manually tested?
1. Clone the repo
   `git clone https://github.com/andela/ah-the-phoenix.git`
2. Change it into the current directory
   `cd ah-the-phoenix`
3. Run tests
   `python manage.py runserver`
4. Register new user on _http://127.0.0.1:8000/api/v1/users/_
Sample Data: 
```
{
  "user": {
    "email": "sample@gmail.com",
    "password": "mypassword",
    "username": "Jacob Maze"
  }
}
```
Sample output: 
image

5. Check email inbox for verification link
Email format:
image

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
 #163388133

### CODE REVIEW

**Comment by dev-jey** [comment]
Hi Paul, I have checked and the feature is working just fine. Keep up the good work.

**Comment by PaulinhoNdegwa** [comment]
> Hi Paul, I have checked and the feature is working just fine. Keep up the good work.

Thank @codact

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -24,9 +31,68 @@ def post(self, request):
...
+
+        try:
+            email = jwt.decode(token, settings.SECRET_KEY)['email']
+            username = jwt.decode(token, settings.SECRET_KEY)['username']
+            user = User.objects.get(email=email)
+
+            if user.is_verified:
+                message = {
+                    'message': 'Account already activated. Click on the link to continue',
+                    'login link': 'http://127.0.0.1:8000/api/v1/users/login'
```
Comment (sirghiny) [inline authors/apps/authentication/views.py]: This won't always be localhost, e.g in the deployed version.

File: authors/apps/authentication/tests/test_signup.py [hunk 2]
```diff
@@ -89,3 +90,41 @@ def test_registeration_for_a_super_user(self):
...
+
+    def test_successful_email_verification(self):
+        """Test if user email can be successfull verified"""
+
+        token = self.signup_a_user(self.user_data).data['token']
+        verification_url = reverse('authentication:verify_email', kwargs={'token':token})
+
+        response = self.client.get(
+            verification_url,
+            HTTP_AUTHORIZATION='token {}'.format(token)
```
Comment (sirghiny) [inline authors/apps/authentication/tests/test_signup.py]: Check out how to use `f` strings in python3, they are much neater for example: https://realpython.com/python-f-strings/

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (PaulinhoNdegwa) [reply authors/apps/authentication/views.py]: Sure. I will edit that.

File: authors/apps/authentication/tests/test_signup.py (hunk 2 above)
Comment (PaulinhoNdegwa) [reply authors/apps/authentication/tests/test_signup.py]: Working on that now.

File: authors/apps/authentication/views.py [hunk 3]
```diff
@@ -24,9 +32,72 @@ def post(self, request):
...
+            'email': email,
+            'username': username,
+            'exp': datetime.utcnow()
+            + timedelta(minutes=60)
+        }
+        token = jwt.encode(payload, settings.SECRET_KEY,
+                           algorithm='HS256').decode('utf-8')
+        sender = os.getenv('EMAIL_HOST_USER')
+        site_link = get_current_site(request)
+        verification_link = 'http://' + site_link.domain + \
```
Comment (sirghiny) [inline authors/apps/authentication/views.py:49]: If you want to use `f` strings, try and be consistent especially in one statement.

**Pull request merged**